### PR TITLE
[dv] Manually update dvsim config files

### PR DESCRIPTION
**Pull request by udinator**
As per the discussion in #840, I've aligned the dvsim config files in `dv/uvm/data` as much as possible with their updated counterparts in the main OpenTitan repository.

Signed-off-by: Udi <udij@google.com>

### CODE REVIEW

**Comment by udinator** [comment]
This force push removes the `sw_build` target from `sim.mk`, as we do not need this for icache simulations.

File: dv/uvm/data/xcelium/xcelium.hjson [hunk 1]
```diff
@@ -7,49 +7,83 @@
 
   // Indicate the tool specific helper sources - these are copied over to the
   // {tool_srcs_dir} before running the simulation.
-  tool_srcs:  ["{proj_root}/dv/uvm/data/xcelium/*"]
+  tool_srcs:  ["{proj_root}/hw/dv/tools/xcelium/*"]
```
Comment (rswarbrick) [inline dv/uvm/data/xcelium/xcelium.hjson:10]: This looks like it might not be quite right?

File: dv/uvm/data/common_sim_cfg.hjson [hunk 2]
```diff
@@ -43,8 +44,13 @@
...
   d: UVM_DEBUG
 
+  // Path to the dut instance (this is used in a couple of places such as coverage cfg
+  // file, xprop cfg file etc. If this is different for your block, then override it with
+  // the 'overrides' directive.
+  dut_instance: "{tb}.dut"
+
   // Default waves dump settings
-  dump_base:  waves
+  dump_file:  waves.{dump}
```
Comment (rswarbrick) [inline dv/uvm/data/common_sim_cfg.hjson:53]: I'd rather keep the dump_base/dump_file stuff as was (see notes at vcs_fsdb.tcl).

**Comment by rswarbrick** [comment]
For what it's worth, I've pushed up a PR with the dumping tidy-ups as https://github.com/lowRISC/opentitan/pull/2128.

File: dv/uvm/data/xcelium/xcelium.hjson (hunk 1 above)
Comment (udinator) [reply dv/uvm/data/xcelium/xcelium.hjson:10]: whoops you're right, good catch

**Comment by udinator** [comment]
@rswarbrick - The missed xcelium directory path is an easy fix, I'll get that pushed up today.
It looks like the remaining comments are regarding the general wave dumping mechanism - since your PR to reflect the code in Ibex is currently under review in OpenTitan, should we wait until lowRISC/opentitan#2128 is merged to reflect those changes here? Otherwise we'll probably have to open a second PR to manually update these files.

**Comment by rswarbrick** [comment]
Let's leave those files unchanged in Ibex until that PR has been merged. I'm also working on some code at the moment that will get rid of lots of the OpenTitan paths from the hjson files. I was hoping it would be done by today, but looks like it will extend into tomorrow. Once that's merged, I'm hoping that we'll genuinely be able to vendor in the code from (OpenTitan's) hw/dv/data directory. We'll have some path changes on the Ibex side, obviously, but that should all come out in the wash, I think.

**Comment by udinator** [comment]
Sounds fine to me; I agree, hopefully we can properly vendor things in soon.
I'll fix the xcelium path and revert the vcs TCL files back to their original state later today (I'm currently attempting to remotely reboot my machine).

**Comment by udinator** [comment]
The latest force pushes fix the Xcelium pathname issue, delete vcs_vpd.tcl, and revert vcs_fsdb.tcl back to its original state.

File: dv/uvm/data/vcs/vcs.hjson [hunk 3]
```diff
@@ -31,13 +31,13 @@
                "-LDFLAGS \"-Wl,--no-as-needed\""]
 
   run_opts:   ["-licqueue",
-               "-ucli -do {tool_srcs_dir}/vcs_fsdb.tcl",
+               "-ucli -do {tool_srcs_dir}/vcs_{dump}.tcl",
```
Comment (rswarbrick) [inline dv/uvm/data/vcs/vcs.hjson:34]: This (confusingly) needs to be `vcs_fsdb.tcl` at the moment.

Comment (udinator) [reply dv/uvm/data/vcs/vcs.hjson]: ah yes good point

**Comment by udinator** [comment]
Latest force push fixes the `vcs_fsdb.tcl` naming inside of `vcs/vcs.hjson`.

**Pull request merged**